src/passes/ConstantPropagation.cpp

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2016 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <pass.h>
+#include <wasm.h>
+
+namespace wasm {
+
+typedef std::map<Name, Const *> State;
```

Comment:
for consistency, i believe we don't have a space in `Const *` elsewhere (we do write `Const *ptr`, but not when the type is by itself). but, maybe this convention is better? i don't feel strongly.